File: Vehicle_ClaimApp/FinalSubmitClaimApp/app/src/main/java/com/example/vehicleinsuranceclaimapp/service/UserSession.java
package com.example.vehicleinsuranceclaimapp.service;

public class UserSession {
    private static UserSession instance;
    private int loggedInUserId;

    private UserSession() {}

    public static synchronized UserSession getInstance() {
        if (instance == null) {
            instance = new UserSession();
        }
        return instance;
    }

    public int getLoggedInUserId() {
        return loggedInUserId;
    }

    public void setLoggedInUserId(int loggedInUserId) {
        this.loggedInUserId = loggedInUserId;
    }
}

File: BookMySlot/app/src/main/java/com/example/service/UserSession.kt
package com.example.service


//This is a user session to track the log in user to display slots based on user
class UserSession private constructor() {

    var loggedInUserId: Int = -1

    companion object {
        @Volatile
        private var instance: UserSession? = null

        fun getInstance(): UserSession {
            return instance ?: synchronized(this) {
                instance ?: UserSession().also { instance = it }
            }
        }
    }
}
